Dynamic resize

gagern opened the pull request:
This is one step along the way towards buttons (#14) since it introduces a `<div>` around our `<canvas>` which can then be used to put in HTML elements. This requires some work to get the size of the `<div>` and the `<canvas>` synchronized properly. That felt like a good opportunity to make that synchronization work not only at initialization time, but also dynamically when the size gets updated.

This is a breaking change, since it is quite possible to specify CSS style sheets where the new DOM tree will render differently than the old one. Some care should be taken when updating to a version including this change, but sane content should be working all right or, in the worst case, quite easy to fix.

montaga commented:
Maybe we should forbid the pattern
`<canvas id="CSCanvas"`
in favor of the preferred `div`-tag?

The FakeCanvas in '/tests/GeoFuncs_tests.js' needs an attribute `setTransform`.

montaga commented:
As far as I understood the changes, they are fine.

But before merging, we should make `node ref/js/runtests.js` work again. In order to fix the testing environment I know to little about how it is built up; so it would mean a lot of afford for me.
@gagern Maybe you could make these changes.

gagern commented:
@montaga thanks for the review. I've addressed the test suite, adjusted the examples and added the forbidden pattern as you suggested. Waiting for OK from Travis, but local `make alltests` succeeded.

gagern commented:
Good thing I waited for Travis, since our master had a few examples missing in my local branch. Rebased and re-converted.

gagern commented:
@montaga Now that Travis is happy, you are happy with the core code change and I addressed the suggestions, I took the liberty of merging this myself. Thanks for looking into this.
